### PR TITLE
config:write important config into metrics

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche <AndreMouche@126.com>

## What have you changed? (mandatory)

For now, we can only find the parameters of the configuration through the log, which is not friendly.
This PR write some important configurations into metrics to fix the above problem.

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

unit tests, integration tests

## Does this PR affect documentation (docs) update? (mandatory)

yes, it will change the metrics.

## Does this PR affect tidb-ansible update? (mandatory)

yes

## Add a few positive/negative examples (optional)

@huachaohuang @rleungx

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -602,6 +711,19 @@ impl DbConfig {
     fn validate(&mut self) -> Result<(), Box<Error>> {
         Ok(())
     }
+
+    fn write_into_metrics(&self) {
```
Comment (zhangjinpeng1987) [inline src/config.rs]: Can we add test for this function?

Comment (AndreMouche) [reply src/config.rs]: Seems can't? Any suggestion?

File: src/raftstore/store/config.rs [hunk 2]
```diff
@@ -358,6 +358,18 @@ impl Config {
...
+
+    pub fn write_into_metrics(&self) {
+        let metrics = register_gauge_vec!(
+            "tikv_config_raftstore",
+            "Config information of raftstore",
+            &["name"]
+        )
+        .unwrap();
+        metrics
+            .with_label_values(&["consistency_check_interval_seconds"])
```
Comment (rleungx) [inline src/raftstore/store/config.rs]: How about other configurations of `raftstore`?

**Comment by huachaohuang** [comment]
We may change some configurations dynamically, so we need to check and report these configurations periodically.

File: src/raftstore/store/config.rs (hunk 2 above)
Comment (AndreMouche) [reply src/raftstore/store/config.rs]: We only output important configurations in this PR.

**Comment by AndreMouche** [comment]
Why do the configurations change dynamically? Can you provide some hints? @huachaohuang

**Comment by huachaohuang** [comment]
We provide some debug API to change configurations dynamically now. For example, tidb-lightning can switch TiKV servers into import mode and then switch them back to normal mode dynamically.

File: src/raftstore/store/config.rs (hunk 2 above)
Comment (rleungx) [reply src/raftstore/store/config.rs]: Yes. But some metrics like `raft_base_tick_interval` may be also important?

**Comment by AndreMouche** [comment]
Where could I find these API docs? @huachaohuang

**Comment by AndreMouche** [comment]
@huachaohuang @rleungx PTAL

**Comment by AndreMouche** [comment]
@huachao PTAL

File: src/config.rs [hunk 3]
```diff
@@ -162,6 +164,114 @@ macro_rules! cf_config {
     };
 }
 
+macro_rules! write_into_metrics {
+    ($cf:expr, $tag:expr,$metrics:expr) => {{
```
Comment (huachaohuang) [inline src/config.rs]:
```suggestion
    ($cf:expr, $tag:expr, $metrics:expr) => {{
```

**Pull request merged**